features/netsocket/emac-drivers/TARGET_STM/mbed_lib.json

```diff
@@ -2,7 +2,8 @@
     "name": "stm32-emac",
     "config": {
         "eth-rxbufnb": 4,
-        "eth-txbufnb": 4
+        "eth-txbufnb": 4,
+        "eth-phyaddr": 0
```

Comment:
Wouldn't it be reasonable to add a `target_override` below for `ARCH_MAX`? Like this:
```
 "target_overrides": {
        ...
        "ARCH_MAX": {
            "eth-phyaddr": 0
        }
    }
```
This way user wouldn't have to figure things out on his/her own.
Also - Some documentation explaining what this option exactly stands for would be a good idea. For example:
```
"config": {
        "eth-rxbufnb": 4,
        "eth-phyaddr": {
            "help" : "Configures the expected value of PHYAD0 pin (0 for pulled-down or 1 for pulled-up",
            "options": [0, 1],
            "value" : 0
        }
```
Feel free to come up with a better description or remove the `option` limitation, if I misunderstood how this pin works.

Reply:
This makes sense, I would change the default for ARCH_MAX to 1 and options can be an array of different values depending on PHY used. DP83848 can use any value between 0 ~ 31. LAN8742 uses only 0 and 1 though.

Reply:
Right, ARCH_MAX should use 1 by default, not 0.
I am not sure if there is a way to define a range of numbers neatly in mbed_lib.json. I haven't found such configuration in any of the existing `mbed_lib.json`s and from json docs I read that you'd have to set a "minimum" and "maximum" values. If you feel like experimenting with this, go ahead, but it's also fine to give up on the "options" field and just specify the valid values in the "help" message.

Reply:
I just took out the options line. If someone is adventurous enough to play with with PHYAD pins, he/she will figure out what value to put there:)